"""
This is a test of the Matrix devide function

>>> with open("7-base_geometry.py", "r") as f:
...     code = f.read()
>>> exec(code) 
>>> bg = BaseGeometry()
>>> dir(bg)
['__class__', '__delattr__', '__dict__', '__dir__', '__doc__', '__eq__', '__format__', '__ge__', '__getattribute__', '__getstate__', '__gt__', '__hash__', '__init__', '__init_subclass__', '__le__', '__lt__', '__module__', '__ne__', '__new__', '__reduce__', '__reduce_ex__', '__repr__', '__setattr__', '__sizeof__', '__str__', '__subclasshook__', '__weakref__', 'area', 'integer_validator']
>>> isinstance(bg, BaseGeometry)
True
>>> bg.integer_validator("myint", 12)
>>> bg.integer_validator("name", "John")
Traceback (most recent call last):
    ...
TypeError: <name> must be an integer
>>> bg.integer_validator("age", 0)
Traceback (most recent call last):
    ...
ValueError: <name> must be greater than 0
>>> bg.integer_validator("age", -4)
Traceback (most recent call last):
    ...
ValueError: <name> must be greater than 0
>>> bg.integer_validator("age", 13.5)
Traceback (most recent call last):
    ...
TypeError: <name> must be an integer
>>> bg.area()
Traceback (most recent call last):
    ...
Exception: area() is not implemented
>>> bg.integer_validator("age")
Traceback (most recent call last):
    ...
TypeError: BaseGeometry.integer_validator() missing 1 required positional argument: 'value'
>>> bg.integer_validator("age", 1)
>>> hasattr(bg, "area")
True
>>> hasattr(bg, "integer_validator")
True
>>> bg.integer_validator("age", "4")
Traceback (most recent call last):
    ...
TypeError: <name> must be an integer
>>> bg.integer_validator("age", (4,))
Traceback (most recent call last):
    ...
TypeError: <name> must be an integer
>>> bg.integer_validator("age", [3])
Traceback (most recent call last):
    ...
TypeError: <name> must be an integer
>>> bg.integer_validator("age", True)
>>> bg.integer_validator("age", {3,4})
Traceback (most recent call last):
    ...
TypeError: <name> must be an integer
>>> bg.integer_validator("age", None)
Traceback (most recent call last):
    ...
TypeError: <name> must be an integer

"""
